app/models/case_management/icasework.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module CaseManagement
+  # Implementation of a CaseManagement for the iCasework Case Management System.
+  class Icasework
+    def initialize(client: nil)
+      @client = client || ::Icasework::Case
+    end
+
+    def submit_foi_request!(name:, email:, body:)
+      request = client.create(
+        request_method: 'Online Form',
+        type: 'InformationRequest',
+        name: name,
+        email: email,
+        details: body
+      )
+
+      SubmittedRequest.new(request)
+    end
+
+    def published_requests(query_params)
```

Comment:
Since this doesn't "work" as is (because of API issues) let's extract this part out into separate commits so that we can merge the submission of requests as soon as its working.